beefy-gadget/src/worker.rs

```diff
@@ -334,26 +283,59 @@ where
 		Ok(sig)
 	}
 
+	/// Return the current active validator set.
+	///
+	/// Note that the validator set could be `None`. This is the case if we don't find
+	/// a BEEFY authority set change and we can't fetch the validator set from the
+	/// BEEFY on-chain state. Such a failure is usually an indication that the BEEFT
+	/// pallet has not been deplyed (yet).
+	fn validator_set(&self, header: &B::Header) -> Option<ValidatorSet<P::Public>> {
+		if let Some(new) = find_authorities_change::<B, P::Public>(header) {
+			Some(new)
+		} else {
+			let at = BlockId::hash(self.client.info().finalized_hash);
```

Comment:
```suggestion
			let at = BlockId::hash(header.hash());
```

I think this will make for much more predictable behavior.